torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -429,10 +431,16 @@ void CudaCodeGen::call(const std::vector<CallArg>& args) {
   std::vector<int> gpu_thread_extents_v(3, 1);
   // evaluate all the block/thread extents into values
   for (int i = 0; i < gpu_block_extents.size(); i++) {
-    gpu_block_extents_v[i] = as_int(gpu_block_extents[i]);
+    auto const& blocks = ExprStmt::make(gpu_block_extents[i]);
+    SimpleIREvaluator eval(blocks, buffer_args());
```

Comment:
Since this code is on critical path, add TODO to:

1. Switch to LLVMCOdeGen
2. Merge the calculation of multiple thread/block extent into a single kernel call with multiple return values.
3. Cache the said kernel.

Reply:
Added a TODO for this; I'm doing a bit of benchmarking now to see how this affects perf at small-ish sizes.

Reply:
But probably not blocking this PR I assume?